src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -73,14 +74,20 @@ struct KeyValueViewModel {
         } else if let groupedBy = param.groupedBy?.name {
             self.init(key: name, value: "\(groupedBy).\(name)")
         } else if param.implementation == .client {
-            self.init(key: name, value: "client.\(name)")
+            self.init(
+                key: name,
+                value: param.required ? "client.\(name)" : name,
```

Comment:
If the implementation is in the client, does it matter whether it is required or not? Wouldn't it still be client.property?

Reply:
Fixed.